emukit/core/loop/user_function.py

```diff
@@ -98,3 +99,41 @@ def evaluate(self, inputs: np.ndarray) -> List[UserFunctionResult]:
         for x, y in zip(inputs, outputs_array):
             results.append(UserFunctionResult(x, y))
         return results
+
+
+class UserFunctionWithCostWrapper(UserFunction):
```

Comment:
I don't think we need this wrapper. Adding optional cost to the existing wrapper does not look too tricky